sdk/core/azure-core/azure/core/pipeline/policies/_utils.py

```diff
@@ -102,3 +102,25 @@ def get_domain(url: str) -> str:
     :return: The domain of the url.
     """
     return str(urlparse(url).netloc).lower()
+
+
+def sanitize_url_query_params(url: str, allowed_query_params: Set[str], placeholder: str = "REDACTED") -> str:
+    """Redact query parameters from the specified URL.
+
+    :param url: The URL to sanitize.
+    :type url: str
+    :param allowed_query_params: The set of query parameters to allow.
+    :type allowed_query_params: set[str]
+    :param placeholder: The placeholder to use for redacted query parameters.
+    :type placeholder: str
+    :rtype: str
+    :return: The sanitized URL.
+    """
+    # Canonicalize the allowed query parameters to lowercase for case-insensitive comparison.
+    lowercase_allowed = {q.lower() for q in allowed_query_params}
```

Comment:
We should avoid normalizing/lowercase the names for each request....

Reply:
Yea, I would like to avoid it. I know our logging policy currently does this, but it can definitely be improved. I initially was thinking of having an internal lowercased set be maintained alongside the public set, however there is some trickiness since several SDKs call `policy.allowed_query_params.update({...})`. Makes it tougher to keep this lowercased set updated.  

How about I introduce a  `CaseInsensitiveSet(Set[str])` class similar to `CaseInsensitiveDict` and just use that?

Reply:
@johanste Added a `CaseInsensitiveSet` class that maintains a lowercased version of the strings for case-insensitive membership checks. From testing, this has similar or slightly better perf depending on the size of the allowlist. This also allows behavior to be the same regardless of if a user does `policy.allowed_query_params.update({...})` or `policy.allowed_query_params = set({...})` (only cosmos does this for header names).